File: core/exercises/03_http_post/cpp/main.cpp
#include <httplib.h>
#include <nlohmann/json.hpp>
#include <cstdlib>
#include <iostream>
#include <string>

int main() {
    const char* env = std::getenv("CODAM_LABS_ECHO_URL");
    // Prefer env (set with --mock). Fallback: https://httpbin.org/post
    (void)env;
    // TODO: POST JSON {"name":"codam"}, print ECHO_OK:codam from response["json"]["name"]
    return 0;
}

File: core/exercises/03_http_post/python/main.py
"""Exercise 03 — HTTP POST."""

import json
import os
import urllib.request


def main() -> None:
    # Prefer CODAM_LABS_ECHO_URL (set automatically with --mock).
    # Live fallback: httpbin — may return 503 / rate-limit; use --mock if so.
    url = os.environ.get("CODAM_LABS_ECHO_URL", "https://httpbin.org/post")
    payload = {"name": "codam"}
    # TODO: POST JSON with Content-Type application/json, print ECHO_OK:<name>
    # Response shape: {"json": {"name": "codam"}, ...}  → use data["json"]["name"]
    pass


if __name__ == "__main__":
    main()
